### ISSUE TITLE
Very long tag name break table

### ISSUE THREAD

**Issue by SimonLab** [dwyl/mvp#217]
![image](https://user-images.githubusercontent.com/6057298/204512775-25896cdd-706d-487c-abf8-2501cc87b189.png)

These are tags created on fly.io when not loggedin
Add a max-width and overflow settings on the tag name (using ellipsis could maybe work?)

**Comment by SimonLab** [comment]
The issue is that the tag text is one word without spaces so it overflow the table width
see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Text/Wrapping_Text

We can see that if we create a long tag name with multiple spaces the tag doesn't overflow:

![image](https://user-images.githubusercontent.com/6057298/204536160-65ead487-3a8f-41ea-a0af-6ae20f3d40f0.png)

**Comment by SimonLab** [comment]
resolved by adding the following classes to the `span` containing the tag name:
`overflow-hidden text-ellipsis whitespace-nowrap inline-block w-40`

see also https://stackoverflow.com/questions/17779293/css-text-overflow-ellipsis-not-working
![image](https://user-images.githubusercontent.com/6057298/204545590-7ccae23d-39a0-4995-bcb3-671362d090d4.png)

**Comment by SimonLab** [comment]
Not there yet:
![image](https://user-images.githubusercontent.com/6057298/204551276-56eafb0c-3c57-4b03-8ab3-3e624b2cce9f.png)

The width is applied to all the tags, it work for long text however for smaller text we want the pill width to match the length of the content, tried to apply  a max width instead but this doesn't work. I need to investigate a bit more how to fix this:

**Comment by SimonLab** [comment]
The max width are working, I wasn't using the correct tailwind class.
However the default max-width are too long:
https://tailwindcss.com/docs/max-width

Luckily Tailwind provides also a way to have a customised class:
![image](https://user-images.githubusercontent.com/6057298/204554916-34a513e4-a242-4d66-aed0-ad6df45acf7a.png)

Going for `max-w-[144px]`

**Comment by nelsonic** [comment]
@SimonLab glad you persisting with this UI quest. 🤞🏼
Please consider using a mobile simulator e.g. Firefox to try and _appear_ mobile-first. 🙏📱

**Comment by SimonLab** [comment]
tested on mobile, see https://github.com/dwyl/mvp/pull/219#discussion_r1034893150

**Comment by nelsonic** [comment]
@SimonLab yes, "Tested on mobile" but screenshots still on desktop ... 🖥️ 
What I mean is: take the screenshot in Firefox with the viewport constrained to a Mobile View e.g: https://github.com/dwyl/dev-setup/issues/55

For a **`new`** person there are no default tags so when they attempt to add tags from the "home" view:

![mvp-no-tags-by-default](https://user-images.githubusercontent.com/194400/204646754-b9ff44db-6bc6-474c-9997-a78fe47b29f1.png)

They need to _manually_ go to the `/tags` page in order to **`Create Tag`**:

![mvp-create-tag](https://user-images.githubusercontent.com/194400/204646777-35ab4510-1410-4db9-ab15-e338b97d85b6.png)

This is undesirable UX. We _definitely_ don't want to force people to change contexts to create a `tag`.
This is a "Why?" moment and should be avoided.

**Comment by SimonLab** [comment]
> "Tested on mobile" but screenshots still on desktop ... desktop_computer
![image](https://user-images.githubusercontent.com/6057298/204749779-1de94c56-d7d5-41d4-a391-214185f243a0.png)

We can see the pills length is based on the tag name, except when the name is too long, then ellipsis are added.

> This is undesirable UX. We definitely don't want to force people to change contexts to create a tag.

I'll open another issue for this.
